zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -361,7 +361,42 @@ where
         }?
     }
 
-    let (tx, tx_metadata) = builder.build(&prover).map_err(Error::Builder)?;
+    let fee_strategy = BasicFixedFeeChangeStrategy::new(DEFAULT_FEE);
+    let balance = fee_strategy
+        .compute_balance(
+            params,
+            target_height,
+            builder.transparent_inputs(),
+            builder.transparent_outputs(),
+            builder.sapling_inputs(),
+            builder.sapling_outputs(),
+        )
+        .map_err(|e| match e {
+            ChangeError::InsufficientFunds {
+                available,
+                required,
+            } => Error::InsufficientBalance(available, required),
+            ChangeError::StrategyError(e) => Error::BalanceError(e),
+        })?;
+
+    for change_value in balance.proposed_change() {
+        match change_value {
+            ChangeValue::Sapling(amount) => {
+                builder
+                    .add_sapling_output(
+                        Some(dfvk.to_ovk(Scope::Internal)),
+                        dfvk.change_address().1,
+                        *amount,
```

Comment:
Why does this need to be dereferenced?

Reply:
`proposed_change` returns a slice, and so it's borrowed here.